client-assertion-validation/src/main/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidator.scala

```diff
@@ -0,0 +1,162 @@
+package it.pagopa.interop.clientassertionvalidation
+
+import cats.data._
+import cats.syntax.all._
+import com.nimbusds.jose.crypto.RSASSAVerifier
+import com.nimbusds.jose.jwk.JWK
+import com.nimbusds.jose.proc.SecurityContext
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier
+import com.nimbusds.jwt.{JWTClaimsSet, SignedJWT}
+import it.pagopa.interop.clientassertionvalidation.Errors._
+import it.pagopa.interop.clientassertionvalidation.model._
+import it.pagopa.interop.commons.utils.TypeConversions._
+import it.pagopa.interop.commons.utils.{DIGEST_CLAIM, PURPOSE_ID_CLAIM}
+
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+trait ClientAssertionValidator {
+  def validateClientAssertion(
+    clientAssertionJws: String,
+    clientId: Option[UUID]
+  ): Either[NonEmptyList[ClientAssertionValidationError], AssertionValidationResult]
+
+  def verifySignature(
+    validationResult: AssertionValidationResult,
+    publicKey: String
+  ): Either[ClientAssertionValidationError, Unit]
+}
+
+final class NimbusClientAssertionValidator(expectedAudience: Set[String]) extends ClientAssertionValidator {
+  private val claimsVerifier: DefaultJWTClaimsVerifier[SecurityContext] =
+    new DefaultJWTClaimsVerifier[SecurityContext](null, null, null, null)
```

Comment:
Given that I don't absolutely remember what the parameters are (maybe the keys?) isn't it possible to locally override the `JWTClaimsVerifier` interface to avoid passing nulls? Will the impl be too verbose?

Reply:
I just noticed that there is an empty constructor `DefaultJWTClaimsVerifier()` which has the same behaviour, may this work for you?